src/api-client.ts

```diff
@@ -5,17 +5,29 @@ import { RequestOptions } from "./types";
 export class ApiClient {
     constructor(private authProvider: IAuthProvider, private apiBaseUrl: string) { }
 
-    public async issueGetRequest(path: string) {
+    public async issueGetRequest(path: string, pageMode: boolean = false) {
```

Comment:
consider creating a separate function for this specific logic

Reply:
The tradeoff here is to duplicate same code as a result of `resolveWithFullResponse: true` on line 17 and write Get func specific for page or edit all func's using `issueGetRequest` and extract body,  BTW `requestPromise` lib deprecated, so we can fix it when we use any replacement lib.
@tomervil, what approach do u think we need to take here?

Reply:
But i totally agree that isn't ideal at all, just a quick win